source_manager.go

```diff
@@ -27,6 +27,18 @@ var sanitizer = strings.NewReplacer(":", "-", "/", "-", "+", "-")
 // sufficient for any purpose. It provides some additional semantics around the
 // methods defined here.
 type SourceManager interface {
+	// SourceManager adds a Release method to the sourceDeducer method set. The
+	// sourceDeducer method set is defined in a separate interface because it's
+	// used by the sourceBridge interface as well, but sourceBridges don't get a
+	// Release method.
+	sourceDeducer
```

Comment:
`Deducer` doesn't really cover it either, though, and we also use that name elsewhere.

My expectation with #159 is actually to depart pretty hard from the `SourceManager` interface with the `*bridge`, as that'll add a bunch of `context.Context` to the methods that there is just no good reason for. 

We can debate the reasons for that over in the other issue when time comes, but for now, it suggests to me that we can leave the public-facing interface here largely unchanged and instead just enumerate the `SourceManager` methods, except for `Release()`, directly in `sourceBridge`.

(that makes sense in my head, does it make sense written out?)